core/scripts/cli/wallet/setDefaultAccountForWallet.js

```diff
@@ -0,0 +1,5 @@
+// Clear out wallet's old accounts and add new account
+module.exports = (web3, defaultAccount) => {
+  web3.eth.accounts.wallet.clear();
+  web3.eth.accounts.wallet.add(defaultAccount.privateKey);
```

Comment:
Does this play well with `truffle`, `truffle-contracts`, and `HDWalletProvider`? Does the provider get discarded or overridden? My understanding is that truffle will effectively ignore the web3 keys if you add them this way. An easier strategy might be to initialize new keys using truffle and HDWalletProvider rather than web3 (not sure if this is possible).

Reply:
greatly simplified the `wallet` module to just read in the account from truffle's HD provider. User can still generate new accounts but we won't custody keys for them

Reply:
Sweet, thanks!